sdk/search/azure-search/src/main/java/com/azure/search/models/FacetResult.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+
+/**
+ * A single bucket of a facet query result. Reports the number of documents
+ * with a field value falling within a particular range or having a particular
+ * value or interval.
+ */
+@Fluent
+public final class FacetResult {
+    /*
+     * Unmatched properties from the message are deserialized this collection
+     */
+    @JsonProperty(value = "")
+    private Map<String, Object> additionalProperties;
+
+    /*
+     * The approximate count of documents falling within the bucket described
+     * by this facet.
+     */
+    @JsonProperty(value = "count", access = JsonProperty.Access.WRITE_ONLY)
+    private Long count;
+
+    /**
+     * Get the additionalProperties property: Unmatched properties from the
+     * message are deserialized this collection.
+     *
+     * @return the additionalProperties value.
+     */
+    public Map<String, Object> getDocument() {
```

Comment:
There's a specific regex replacing getAdditionalProperties with getDocument for FacetResult.java, SearchResult.java and SuggestResult.java post a review on usabilty with external reviewers. azure-core names the document ressult as additionalProperties, which was not usable. 
@brjohnstmsft please advice if this should change. If so, it will require a change to the swagger  config as well

Reply:
@navalev The name is incorrect in this particular case, because the returned object is not actually a document in any sense of the word. The name is fine for `SearchResult` and `SuggestResult`, because in those cases it's actually a document.

Ideally this property wouldn't even need to be public -- it should only be consumed from `ValueFacetResult` and `RangeFacetResult`, and there should be methods on `FacetResult` to create one of those for a given type parameter, like in .NET. However, I realize that deviates pretty far from what the code generator can produce. Perhaps it will require deleting the generated file and substituting a hand-written one.

I suggest we handle this as part of the maintenance backlog, in addition to re-doing the API review for faceting scenarios. We will need to do that before shipping a preview. There's no need to block the PR on this.